File: appJava/src/main/java/com/brandon/appjava/di/NavigationModule.java
package com.brandon.appjava.di;

import com.brandon.appjava.navigation.LegacyNavigationImplJava;
import com.brandon.corejava.navigation.LegacyNavigationJava;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.android.components.ViewModelComponent;

@Module
@InstallIn(ViewModelComponent.class)
public class NavigationModule {

    @Provides
    public LegacyNavigationJava providesNavigation() {
        return new LegacyNavigationImplJava();
    }

}

File: appLegacy/src/main/java/com/brandon/di/NavigationModule.kt
package com.brandon.di

import com.brandon.navigation.LegacyNavigation
import com.brandon.navigation.LegacyNavigationImpl
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ViewModelComponent

@Module
@InstallIn(ViewModelComponent::class)
class NavigationModule {

    @Provides
    fun providesLegacyNavigation(): LegacyNavigation {
        return LegacyNavigationImpl()
    }

}